[Feature] Audio Player: Add Album to Queue (#2)
nawhas/rfcs: feature.add-album-to-queue.md

# Audio Player: Add Album to Queue
**Authors:** [@szainmehdi](https://github.com/szainmehdi) [@shea786](https://github.com/shea786)  
**Type:** Feature

## Overview
Allow the user to add the entire album to the audio player queue

## Why are we making this change?

> So that it is easier for the user, instead of them going to each track to add to queue, one button would add the entire track

Reviewer: ```suggestion
Playing an entire album is a common feature users expect and a feature Nawhas.com is currently lacking.
```

## Requirements
- User should be able to an entire album to the playlist

> - The order of which the tracks are being displayed is the order of which the tracks should be added to the queue

Reviewer: ```suggestion
- The order in which the tracks are being displayed is the order in which the tracks should be added to the queue
```

> - User should be able to add all tracks from the `Reciter Profile` page against each table to the queue

Reviewer: I'm unsure what you mean here. Can you clarify?

Author: I guess I didnt explain it right, my bad

basically against each album on the reciter page, we want to show a button to play the all tracks from the album, re-worded in the new commit

> - User should be able to add all the tracks from the `Tracks Show` page to the queue

Reviewer: ```suggestion
- On the Track page: the "More From This Album" card should allow users to play all tracks from the album
```

> - When adding all tracks from an album to the queue, if the queue is shuffled then add the new tracks to the shuffled queue in random order

Reviewer: I don't like this idea.

In fact, I have something else in mind. 

I think the right approach here is:

- A `play` button is shown on albums.
- A `play` button is shown on the "More From This Album" card.
- Clicking the `play` button discards the current queue and shuffle state, and plays the selected album from top to bottom.

Author: I don't think we should be doing that,

In my view, if we have gone through and added tracks to our queue, and later we click play against an album, our current queue will change to the album tracks,
I personally would get really annoyed after it took me so long to add tracks to my queue for it to get overwritten. 

If users don't want the current tracks from the current queue, they can always go and remove them manually right?

Author: Maybe as per your mockups, we should have two buttons,

Play which would overwrite the queue
And add to queue which would add as per my logic above

That would be better I think

## Detailed Eng Design
### API
No changes made to API

### Frontend
#### Reciter Profile
  - Add a new button against the album table to add all tracks to the album
  - Add a new method named `addAlbumToQueue` which will commit a new mutation on the `player` store to add the tracks to the queue

#### Tracks Page
  - Add a new button above the tracks from this album to add album to the queue
  - Add a new method named `addAlbumToQueue` which will commit a new mutation on the `player` store to add the tracks to the queue

- Proposed button name for `Reciter Profile` and `Tracks Page` - `Add Album to Queue`

#### Vuex
##### Player Store

>   -Adding a new mutation named `ADD_ALBUM_TO_QUEUE`

Reviewer: ```suggestion
  - Adding a new mutation named `ADD_ALBUM_TO_QUEUE`
```

  - This new mutation will accept an array of tracks
  - Will loop through each track and add the track to the queue
  - If the queue is currently shuffled
    - Pass the list of racks received to the `shuffle` function to generate a random order
    - Then push the shuffled array to the `shuffled` queue

### Migration Path
No changes to migration

### Deployment Strategy

> ##Mockups

Reviewer: ```suggestion
## Mockups
```
